src/test/java/com/teragrep/cfe_39/CombinedFullTest.java

```diff
@@ -0,0 +1,309 @@
+import org.apache.hadoop.fs.Path;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.sql.Timestamp;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class CombinedFullTest {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CombinedFullTest.class);
+    private static MiniDFSCluster hdfsCluster;
+    private static File baseDir;
+    private static Config config;
+
+
+    // Start minicluster and initialize config.
+    @BeforeAll
+    public static void startMiniCluster() throws IOException, InterruptedException {
+        config = null;
+        try {
+            config = new Config();
+        } catch (IOException e){
+            LOGGER.error("Can't load config: {}", e.toString());
+            Assertions.fail();
+        } catch (IllegalArgumentException e) {
+            LOGGER.error("Got invalid config: {}", e.toString());
+            Assertions.fail();
+        }
+        // Create a HDFS miniCluster
+        baseDir = Files.createTempDirectory("test_hdfs").toFile().getAbsoluteFile();
+        Configuration conf = new Configuration();
+        conf.set(MiniDFSCluster.HDFS_MINIDFS_BASEDIR, baseDir.getAbsolutePath());
+        MiniDFSCluster.Builder builder = new MiniDFSCluster.Builder(conf);
+        hdfsCluster = builder.build();
+        String hdfsURI = "hdfs://localhost:"+ hdfsCluster.getNameNodePort() + "/";
+        // System.out.println("hdfsURI: " + hdfsURI);
+        config.setHdfsuri(hdfsURI);
+        DistributedFileSystem fileSystem = hdfsCluster.getFileSystem();
+    }
+
+    // Teardown the minicluster
+    @AfterAll
+    public static void teardownMiniCluster() {
+        hdfsCluster.shutdown();
+        FileUtil.fullyDelete(baseDir);
+    }
+
+    @Test
+    public void kafkaAndAvroFullTest() throws InterruptedException, IOException {
+        insertMockFiles(); // Maybe add modifier that allows adding or not adding the mock file.
+        config.setMaximumFileSize(3000); // 10 loops (140 records) are in use at the moment, and that is sized at 36,102 bytes.
+        KafkaController kafkaController = new KafkaController(config);
+        Thread.sleep(10000);
+        kafkaController.run();
+        // The avro files should be committed to HDFS now. Check the committed files for any errors.
+        // There should be 20 files, 10 partitions with each having 2 files assigned to them.
+        // hdfsReadCheck(); does not work properly if pruning is enabled and prune offset is set too low, which causes the records to be pruned from the database.
+        if (config.getPruneOffset() == 157784760000L) {
+            try {
+                hdfsReadCheck();
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+    }
+    @Test
+    public void hdfsPruneTest() throws IOException {
```

Comment:
Tests should not throw nor try/catch anything, use `Assertions.assertDoesNotThrow`, `Assertions.assertThrows`, `Assertions.assertAll` and such instead

Reply:
Solved from commit 65342c2e865912f981dfe82f470cab6884f72cdb onward